pyresample/_spatial_mp.py

```diff
@@ -110,36 +110,15 @@ def query(self, x, k=1, eps=0, p=2, distance_upper_bound=np.inf):
 
 
 class BaseProj(pyproj.Proj):
+    """Helper class for easier backwards compatibility."""
 
     def __init__(self, projparams=None, preserve_units=True, **kwargs):
-        # Copy dict-type arguments as they will be modified in-place
-        if isinstance(projparams, dict):
-            projparams = projparams.copy()
-
-        # Pyproj<2 uses __new__ to initiate data and does not define its own __init__ method.
         if is_pyproj2():
-            # If init is found in any of the data, override any other area parameters.
-            if 'init' in kwargs:
-                warnings.warn('init="EPSG:XXXX" is no longer supported. Use "EPSG:XXXX" as a proj string instead')
```

Comment:
The deprecation warnings now come directly from pyproj, right?

Reply:
That is the hope, yes. It is still an opt-in "feature", but there is a new issue/PR to add this to the next version of pyproj so that users see the deprecation sooner.